monai/transforms/spatial/array.py

```diff
@@ -173,11 +179,11 @@ def __call__(
             affine_ = np.eye(sr + 1, dtype=np.float64)
         else:
             affine_ = to_affine_nd(sr, affine)
-        out_d = self.pixdim[:sr]
-        if out_d.size < sr:
-            out_d = np.append(out_d, [1.0] * (out_d.size - sr))
-        if np.any(out_d <= 0):
-            raise ValueError(f"pixdim must be positive, got {out_d}.")
+        if not issequenceiterable(self.pixdim):
+            out_d = ensure_tuple_rep(self.pixdim, sr)
```

Comment:
by default we should assume pixdim 1.0 instead of `ensure_tuple_rep`

Reply:
So I think you mean we should treat `pixdim=0.5` and `pixdim=[0.5]` as the same? And if applying on a 3D data, it should map to `pixdim=[0.5, 1.0, 1.0]` instead of `pixdim=[0.5, 0.5, 0.5]` or `pixdim=[0.5, <image pixdim[1]>, <image pixdim[2]>]`?

Thanks.

Reply:
Hi @wyli , could you please help confirm which the expected behavior is? Then I will update the PR ASAP.

Thanks.

Reply:
updated in the latest commit.
Thanks.